src/PowerShellAPI.cs

```diff
@@ -0,0 +1,144 @@
+            {
+                foreach (var mod in moduleInfo)
+                {
+                    foreach (var cmdletInfo in mod.ExportedCommands.Values)
+                    {
+                        if (cmdletInfo.CommandType != CommandTypes.Alias)
+                        {
+                            cmdletInfos.Add(cmdletInfo);
+                        }
+                    }
+                }
+            }
+            else
+            {
+                throw new ItemNotFoundException(moduleName);
+            }
+
+            return cmdletInfos;
+        }
+
+        internal static Collection<PSModuleInfo>? GetModuleInfo(string moduleName, PSSession? session = null)
+        {
+            System.Management.Automation.PowerShell selectedPS = GetPowerShell(session);
+
+            Collection<PSModuleInfo>? modules = null;
+
+            try
+            {
+                // first try if the module is loaded
+                modules = selectedPS
+                    .AddCommand(@"Microsoft.PowerShell.Core\Get-Module")
+                    .AddParameter("Name", moduleName)
+                    .Invoke<PSModuleInfo>();
+
+                if (modules.Count == 0)
+                {
+                    // if not found try to import first
+
+                    selectedPS.Commands.Clear();
+
+                    modules = selectedPS
+                        .AddCommand(@"Microsoft.PowerShell.Core\Import-Module")
+                        .AddParameter("Name", moduleName)
+                        .AddParameter("PassThru")
+                        .Invoke<PSModuleInfo>();
+                }
+            }
+            // swallow the exceptions and eventually return null;
+            catch (FileNotFoundException)
+            {
+            }
+            catch (System.Management.Automation.ActionPreferenceStopException)
+            {
+            }
+
+            return modules;
+        }
+
+        internal static Collection<PSObject> GetHelpForCmdlet(string cmdletName, PSSession? session = null)
+        {
+            System.Management.Automation.PowerShell selectedPS = GetPowerShell(session);
+
+            return selectedPS
+                .AddCommand(@"Microsoft.PowerShell.Core\Get-Help")
+                .AddParameter("Name", cmdletName)
+                .AddParameter("Full")
+                .Invoke();
+        }
+
+        private static System.Management.Automation.PowerShell GetPowerShell(PSSession? session)
+        {
+            System.Management.Automation.PowerShell selectedPS;
+
+            if (session is null)
+            {
+                ps ??= System.Management.Automation.PowerShell.Create(RunspaceMode.CurrentRunspace);
+                ps.Commands.Clear();
+                selectedPS = ps;
+            }
+            else
+            {
+                selectedPS = System.Management.Automation.PowerShell.Create();
+                selectedPS.Runspace = session.Runspace;
```

Comment:
Does disposing a `PowerShell` object, also dispose the assigned runspace? If that is the case, we cannot Dispose as we do not own the PSSession. If it does not then we can dispose.

In case of using current runspace, we want to re-use the `PowerShell` object.

Your thoughts?